Doc/library/inspect.rst

```diff
@@ -503,6 +503,7 @@ Retrieving source code
    If the documentation string for an object is not provided and the object is
    a class, a method, a property or a descriptor, retrieve the documentation
    string from the inheritance hierarchy.
+   Might return ``None`` for invalid docstrings.
```

Comment:
This method returns `None` if the object doesn't have a docstring. I don't think a docstring even could be invalid.

I'd actually prefer this method returning an empty string if the object doesn't have a docstring. Needing to use `doc = getdoc(x) or ''` everytime is annoying but I can understand changing this would be a backwards incompatible change.

Reply:
I've updated this to be: ` Might return None for missing or invalid docstrings`

I mean this case by `invalid`:

```python
    if not isinstance(doc, str):
        return None
```